### PR TITLE
Fix Joint3D and Joint2D warning causing a crash

### LINKED ISSUE #55660: Editor crashed when closing a scene using a PinJoint2D
### Godot version

e1e3ea59785b58fc74cde44f4ea47a561a3ca293

### System information

Linux binogure 5.15.0-2-amd64 #1 SMP Debian 5.15.5-1 (2021-11-26) x86_64 GNU/Linux

### Issue description

Some scenes can be opened and closed without an inch. However some scenes once opened cannot be closed without crashing the Editor.

### Steps to reproduce

1. Clone this repository: https://github.com/binogure-studio/real-estate-online (git commit: 494d0e0 )
2. Open the game folder as a Godot project
3. Open `res://scene/classic/wheel.tscn`
4. Close the scene

### LINKED ISSUE #60962: Editor crash when closing scene with Joint3D
### Godot version

v4.0.alpha7.official

### System information

Kubuntu 20.04, Intel i5-2500, 16Go, NVIDIA GTX1650

### Issue description

If a scene contains a `Joint3D` node (PinJoint3D, HingeJoint3D, ...), this scene cannot be closed from the editor without crashing the Godot engine.

### Steps to reproduce

Create a scene with `Node3D` root. Add a `PinJoint3D` node into this scene.
Now if we close the scene by clicking on the cross of the tab, the Godot engine crashes.
img1

img2

### Minimal reproduction project

[IssueCrashCloseScene.zip](https://github.com/godotengine/godot/files/8676095/IssueCrashCloseScene.zip)

### PR DESCRIPTION
**Pull request by trollodel**
Fixes #60962. 
Fixes #55660. 
Introduced by #48242.

This PR fixes the crash, but it's hiding a worse problem.
Joint3D (the same happens for Joint2D) makes the offending `update_configuration_warnings` call when it's handling the `NOTIFICATION_EXIT_TREE` notification. The warning signal is received by SceneTreeEditor with the signal connected [here](https://github.com/godotengine/godot/blob/3568b3deeaf3fbbf9e728ff25c7d0349075f9f1a/editor/scene_tree_editor.cpp#L717). The `connect` call has the `CONNECT_DEFERRED` flag, which means the signal may be handled after the Joint3D is freed. Removing that flag doesn't trigger the crash.
As a plus, `VariantCasterAndValidate` doesn't check if the object is valid, but this may be expected.
In conclusion, any node that calls `update_configuration_warnings` and right after it's freed can cause a crash.

### CODE REVIEW

**Comment by akien-mga** [comment]
CC @rburing

**Comment by akien-mga** [comment]
I assume it's relevant for `3.x` as #58642 was a backport of #58641?

Also, #55660 predates the merge of #58642 so it's a bit confusing, was it really caused by this PR?

**Comment by trollodel** [comment]
> Also, https://github.com/godotengine/godot/issues/55660 predates the merge of https://github.com/godotengine/godot/pull/58642 so it's a bit confusing, was it really caused by this PR?

Yeah, I read the diff wrongly. That code was already there before that PR. 
It was introduced by #48242 and a 3.x versions isn't needed, if I'm right about the cause.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!
